plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -2691,20 +2722,25 @@ else if (strings.size() != 2) {
             String schema = strings.get(0);
             String name = strings.get(1);
             SchemaTableName schemaTableName = new SchemaTableName(schema, name);
-            IcebergTableHandle tableHandle = getTableHandle(session, schemaTableName, Optional.empty(), Optional.empty());
+            ConnectorTableHandle tableHandle = getTableHandle(session, schemaTableName, Optional.empty(), Optional.empty());
 
             if (tableHandle == null) {
                 // Base table is gone
                 return new MaterializedViewFreshness(STALE, Optional.empty());
             }
+
+            if (tableHandle instanceof CorruptedIcebergTableHandle) {
```

Comment:
combine with `if (tableHandle == null) {` preceding case (updating the comment message inside)